client/src/boot/BootRoutes.tsx

```diff
@@ -187,15 +183,15 @@ class BootRoutes {
    *
    * @return {Boolean}
    */
-  shouldConfirmBeforeUnload() {
+  private shouldConfirmBeforeUnload() {
     const state = this.store.getState();
     const forms = state.unsavedForms || [];
     const schemas = state.form.formSchemas;
 
     const changedForms = forms.filter((form) => {
-      const schema = Object.values(schemas).find(item => item.name === form.name);
+      const schema = Object.keys(schemas).find(item => item['name'] === form.name);
 
-      const notify = schema && schema.state && schema.state.notifyUnsavedChanges;
+      const notify = schema && schema['state'] && schema['state'].notifyUnsavedChanges;
```

Comment:
This is a somewhat annoying aspect of TS. Sometimes you are pass an object with an undefined structure. If you want to access some of those object's properties, you have to either cast the object has any, or use a PHP-style square bracket syntax.